testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViews.java

```diff
@@ -13,5 +13,71 @@
  */
 package io.trino.tests.hive;
 
+import io.trino.tempto.Requires;
+import io.trino.tempto.assertions.QueryAssert;
+import io.trino.tempto.fulfillment.table.hive.tpch.ImmutableTpchTablesRequirements;
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryExecutor.query;
+import static io.trino.tests.TestGroups.HIVE_VIEWS;
+import static io.trino.tests.utils.QueryExecutors.onHive;
+import static io.trino.tests.utils.QueryExecutors.onPresto;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@Requires({
+        ImmutableTpchTablesRequirements.ImmutableNationTable.class,
+        ImmutableTpchTablesRequirements.ImmutableOrdersTable.class,
+})
 public class TestHiveViews
-        extends AbstractTestHiveViews {}
+        extends AbstractTestHiveViews
+{
+    @Test(groups = HIVE_VIEWS)
+    public void testFailingHiveViewsForInformationSchema()
+    {
+        // The expected behavior is different across hive versions. For hive 3, the call "getTableNamesByType" is
+        // used in ThriftHiveMetastore#getAllViews. For older versions, the fallback to doGetTablesWithParameter
+        // is used, so information_schema.views does not include translated Hive views.
+
+        onHive().executeQuery("DROP SCHEMA IF EXISTS test_list_failing_views CASCADE");
+        onHive().executeQuery("CREATE SCHEMA test_list_failing_views");
+        onHive().executeQuery("CREATE VIEW test_list_failing_views.correct_view AS SELECT * FROM nation limit 5");
+
+        // Create a view for which the translation is guaranteed to fail
+        onPresto().executeQuery("CREATE TABLE test_list_failing_views.table_dropped (col0 BIGINT)");
+        onHive().executeQuery("CREATE VIEW test_list_failing_views.failing_view AS SELECT * FROM test_list_failing_views.table_dropped");
+        onPresto().executeQuery("DROP TABLE test_list_failing_views.table_dropped");
+
+        runQueryWithAssertion(
+                "SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views'",
+                getHiveVersionMajor() == 3
+                        ? queryAssert -> queryAssert.containsOnly(row("correct_view"))
+                        : QueryAssert::hasNoRows);
+
+        runQueryWithAssertion("SELECT table_name FROM information_schema.views",
+                getHiveVersionMajor() == 3
+                        ? queryAssert -> queryAssert.contains(row("correct_view"))
+                        : queryAssert -> assertThatThrownBy(() -> queryAssert.contains(row("correct_view"))));
```

Comment:
`assertThatThrownBy` should be followed with the actual assertion.

Reply:
nit; can we make it 
```
if ( getHiveVersionMajor() == 3) {
// assertThat(...)
}
else {
// different assertThat...
}

Reply:
@losipiuk @findepi 

Removed ternary conditional operator to simplify (and hence also removed the related refactoring commit)

So, previously the PR had: 

```
        runQueryWithAssertion(
                "SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views'",
                getHiveVersionMajor() == 3
                        ? queryAssert -> queryAssert.containsOnly(row("correct_view"))
                        : QueryAssert::hasNoRows);

        runQueryWithAssertion(
                "SELECT table_name FROM information_schema.views",
                getHiveVersionMajor() == 3
                        ? queryAssert -> queryAssert.contains(row("correct_view"))
                        : queryAssert -> assertThatThrownBy(() -> queryAssert.contains(row("correct_view"))));
```
 
and now changed it to

```
        String withSchemaFilter = "SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views'";
        String withNoFilter = "SELECT table_name FROM information_schema.views";
        if (getHiveVersionMajor() == 3) {
            assertThat(query(withSchemaFilter)).containsOnly(row("correct_view"));
            assertTrue(query(withNoFilter).rows().contains(ImmutableList.of("correct_view")));
        }
        else {
            assertThat(query(withSchemaFilter)).hasNoRows();
            assertFalse(query(withNoFilter).rows().contains(ImmutableList.of("correct_view")));
        }
```

One problem is that `QueryAssert` doesn't have `doesNotContain` method currently, so ended up using `assertTrue` and `assertFalse` for the `withNoFilter` query. LMK if you'd prefer `assertThatThrownBy` (which seems bit hard to read) or any other cleaner way to handle this.

Reply:
Can't you assert on `rows()` using standard collection assertion:

```java
Assertions.assertThat(query(withNoFilter).rows()).doesNotContain(.....)
```